cognite/neat/core/extractors/rdf_to_relationships.py

```diff
@@ -352,17 +355,28 @@ def _micro_batch_push(
     """
     total = len(relationships)
     counter = 0
+    if push_type not in ["update", "create"]:
+        logging.info(f"push_type {push_type} not supported")
+        raise ValueError(f"push_type {push_type} not supported")
 
     for batch in chunker(relationships, batch_size):
         counter += len(batch)
         start_time = datetime_utc_now()
-        if push_type == "update":
-            client.relationships.update(batch)
-        elif push_type == "create":
-            client.relationships.create(batch)
-        else:
-            logging.info(f"push_type {push_type} not supported")
-            raise ValueError(f"push_type {push_type} not supported")
+        retry_counter = 0
+        while True:
+            try:
+                if push_type == "update":
+                    client.relationships.update(batch)
+                elif push_type == "create":
+                    client.relationships.create(batch)
+                break
+            except Exception as e:
+                if retry_counter >= max_retries:
+                    raise e
+                retry_counter += 1
+                logging.error(f"Error pushing {push_type} micro-batch: {e}")
+                logging.info(f"Retrying ({retry_counter})  in {retry_delay} seconds")
+                time.sleep(retry_delay)
```

Comment:
I think there should be a function/decorator, to handle this logic, to avoid repeating it everywhere.

Reply:
fixed